### PR TITLE
move extension to ./pgmq-extension

### PR DESCRIPTION
**Pull request by ChuckHend**
Moves all the extension files to `./extension`. This will hopefully help keep the overall project (pg extension, client libraries) more organized. This should hopefully make it easier to separate the various CI jobs for each sub-project in this repo.

### CODE REVIEW

**Comment by ChuckHend** [comment]
I'm thinking maybe wait to merge this until after @theory PRs merge so that there are not massive merge conflicts.

**Review by v0idpwn — APPROVED** [review]

**Comment by v0idpwn** [comment]
Maybe overthinking but could call it `pgmq-extension`, envisioning possible extra extensions in the future, like a different one for topics/routing/fan-out, or bgw/rust, etc

**Comment by ChuckHend** [comment]
> pgmq-extension

Good idea. I'll push that up.

**Review by v0idpwn — APPROVED** [review]

**Review by v0idpwn — APPROVED** [review]

**Pull request merged**
